src/libtls/tls_server.c

```diff
@@ -1159,6 +1159,23 @@ static status_t send_server_hello(private_tls_server_t *this,
 	}
 	else
 	{
+		if (this->tls->get_version_max(this->tls) == TLS_1_3 &&
+			version < TLS_1_3)
```

Comment:
This can never be true because `version` is set to `this->tls->get_version_max()` just above. So this patching of `server_random` should probably happen in `process_client_hello()` when we still can determine what the original max. version was (e.g. store it in a local variable before the `if (versions.len)` check and add this fixup after determining/setting the version).